pkg/kubelet/dockertools/exec.go

```diff
@@ -90,15 +88,22 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		return command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
+	}
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
```

Comment:
We should be running the timer in a separate go thread and stop that gothread/timer via a defer statement. IIUC, this current logic will not cause a timeout at all.